src/hints/dict/felt252DictEntryInit.ts

```diff
@@ -0,0 +1,51 @@
+import { z } from 'zod';
+
+import { Felt } from 'primitives/felt';
+import { VirtualMachine } from 'vm/virtualMachine';
+
+import { resOperand, ResOperand } from 'hints/hintParamsSchema';
+import { HintName } from 'hints/hintName';
+
+/** Zod object to parse Felt252DictEntryInit hint */
+export const felt252DictEntryInitParser = z
+  .object({
+    Felt252DictEntryInit: z.object({
+      dict_ptr: resOperand,
+      key: resOperand,
+    }),
+  })
+  .transform(({ Felt252DictEntryInit: { dict_ptr, key } }) => ({
+    type: HintName.Felt252DictEntryInit,
+    dictPtr: dict_ptr,
+    key,
+  }));
+
+/**
+ * Felt252DictEntryInit hint
+ *
+ * Initialize the entry to be added to a given dictionnary.
+ */
+export type Felt252DictEntryInit = z.infer<typeof felt252DictEntryInitParser>;
+
+/**
+ * Get the dictionnary `dict` at `dictPtr`
+ * Get the `key` to be initialized
+ *
+ * Read the current value at `key` if any,
+ * Initialize `dict[key]` to 0 if undefined.
+ *
+ * Assert the value at `key` to the `dict` segment.
+ *
+ */
+export const felt252DictEntryInit = (
+  vm: VirtualMachine,
+  dictPtr: ResOperand,
+  key: ResOperand
+) => {
+  const address = vm.getPointer(...vm.extractBuffer(dictPtr));
+  const keyValue = vm.getResOperandValue(key).toString();
+  const dict = vm.getDict(address);
+  const prevValue = dict.get(keyValue) || new Felt(0n);
+  dict.set(keyValue, prevValue);
+  vm.memory.assertEq(address.add(1), prevValue);
```

Comment:
`1` should be a named const

Reply:
`address` is a pointer to the next dict access: for a dictionnary on segment 8, for the first key initialized, `address == new Relocatable(8, 0)` is (`8:0`) ; next entry `address` will be `8:3` etc

A dictionnary access works by block of 3 cells:
- First cell is the key
- Second cell is the previous value, initialized at 0 if undefined --> `Felt252DictEntryInit` assert this value in memory
- Third cell is the current value --> `Felt252DictEntryUpdate` assert this value in memory

Naming `1` would not make much sense here, I'd rather name `address.add(1)` as `prevValueAddress`.
Though leaving it inlined is better imo

Reply:
key_offset is 0, prev_value_offset is 1, new_value_offset is 2, I still think that something like `vm.memory.assertEq(address.add(PREV_VALUE_OFFSET), prevValue);` would be better as it conveys in a single word the entire explanation given above

Reply:
legit, will refactor Dictionary and related constants to their own file `dictionary.ts`